cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1374,22 +1387,99 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
+                            } else if (desiredReplicas < currentReplicas) {
+                                zkScalingDown = true;
+                            }
+
+                            ConditionBuilder zkManualScalingCondition = null;
+
+                            if ((zkScalingUp || zkScalingDown) && isZK35x) {
+                                // This is a scale up or down of a ZK 3.5.x cluster
+
+                                String scaleMessage = String.format(
+                                        "Scaling Zookeeper 3.5.x cluster: Rolling updates will be skipped and the manual " +
+                                        "update process defined in the documentation should be followed. When finished set " +
+                                         "%s annotation on the Zookeeper StatefulSet to false.",
+                                        Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE);
+
+                                log.info(scaleMessage);
 
+                                zkManualScalingCondition = new ConditionBuilder()
+                                        .withLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(dateSupplier()))
+                                        .withType("ZK-Manual-Scaling")
+                                        .withReason("replicaChange")
+                                        .withStatus("True")
+                                        .withMessage(scaleMessage);
+
+                                // Setting this will add the annotation to the STS in the state.zkStatefulSet() method
+                                zkManualScaling = true;
+
+                            } else if (Annotations.booleanAnnotation(sts, Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE, false)) {
+                                // This is an on-going manual scaling operation. Once the annotation has been set to true by the operator
+                                // only the user can set it to false.
+                                String scaleMessage = String.format("Detected manual Zookeeper scaling operation in progress. " +
+                                        "Rolling of Zookeeper cluster will remain disabled until %s annotation is set to false.",
+                                        Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE);
+
+                                log.info(scaleMessage);
+
+                                zkManualScalingCondition = new ConditionBuilder()
+                                        .withLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(dateSupplier()))
+                                        .withType("ZK-Manual-Scaling")
+                                        .withReason("manualScaleAnnotationSet")
+                                        .withStatus("True")
+                                        .withMessage(scaleMessage);
+
+                                zkManualScaling = true;
+
```

Comment:
I don't like nitpicking about empty lines, but it really hits me in the eyes that half of the code has empty line before `}` and the other half doesn't.